### PR TITLE
COST-2896: GCP: extend date ranges for monthly views

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-2896](https://issues.redhat.com/browse/COST-2896)

## Description

This change will ...

## Testing
Upload the dev gcp account. 

1. http://127.0.0.1:5042/api/cost-management/v1/gcp_invoice_monthly_cost/?provider_uuid={{source_uuid}}
2. http://localhost:8000/api/cost-management/v1/reports/gcp/costs/?filter[time_scope_units]=month&filter[resolution]=monthly&filter[time_scope_value]=-1
3. http://localhost:8000/api/cost-management/v1/reports/gcp/costs/?filter[time_scope_units]=month&filter[resolution]=monthly&filter[time_scope_value]=-2

These costs should match

## Notes

...

### CODE REVIEW

**Comment by myersCody** [comment]
Smoke Tests: https://ci.ext.devshift.net/job/project-koku-koku-pr-check/6355/console

**Comment by myersCody** [comment]
There are a 19 test failures with OCP on GCP, I know Luke mentioned there was something going on with those tests so I will sync with him tomorrow.

File: koku/api/query_handler.py [hunk 1]
```diff
@@ -320,23 +320,24 @@ def _get_gcp_filter(self, delta=False):
...
         else:
             start = self.start_datetime
             end = self.end_datetime
-        start_filter = QueryFilter(field="usage_start", operation="gte", parameter=start.date())
-        end_filter = QueryFilter(field="usage_start", operation="lte", parameter=end.date())
+            invoice_months = self.dh.gcp_find_invoice_months_in_date_range(start.date(), end.date())
+            # We add a 5 day buffer to the start & end here
+            # to handle cross over data. We may want to change
+            # this to an env in the future
+            if self.parameters.get_filter("time_scope_value") and self.time_scope_value in [-1, -2, -3]:
```
Comment (lcouzens) [inline koku/api/query_handler.py:331]: Technically we don't need to add to the end date of -1 since that's today 😜

**Review by lcouzens — APPROVED** [review]

**Comment by lcouzens** [comment]
Smoke look great on this run so we can merge this now.

**Pull request merged**